### DISCUSSION TITLE
If a wheel is in contact with the ground, can I directly obtain the Jacobian and position information of the contact point using Pinocchio?

### DISCUSSION THREAD

**Discussion by lkx8421** [stack-of-tasks/pinocchio#2306] [Q&A]
If a wheel is in contact with the ground, can I directly obtain the Jacobian and position information of the contact point using Pinocchio?

**Comment by umairdotdev** [comment]
Yes, it is possible to do that. You can define a frame at the contact point and get position information and Jacobaian for this frame.

**Reply by lkx8421** [reply]
But wheel contact point is not fixed in the wheel,

**Reply by jorisv** [reply]
I think you can safely update the frame placement by updating the right frame in the frames vector: `model.frames[i].placement`

Also, you can also call [https://github.com/stack-of-tasks/pinocchio/blob/master/include/pinocchio/algorithm/frames-derivatives.hpp#L14-L48](getFrameVelocityDerivatives) that take placement as an argument.
